syntax = "proto3";

package task_service;

option go_package = "github.com/j-haj/bdl/task_service";

service TaskService {
  rpc RequestTask(TaskRequest) returns (Task) {}
}

message Task {
  // task_id should be the source address followed by an identifier
  // for example "localhost:1000#1"
  string task_id = 1;
  string source = 2;
  bytes task_obj = 3;
}

message TaskRequest {}
message ResultResponse {}
